shared-libs/transitions/test/unit/pregnancy_registration.js

```diff
@@ -401,3 +401,203 @@ describe('pregnancy registration', () => {
+          trigger: 'add_expected_date',
+          params: '',
+          bool_expr: 'typeof doc.getid === "undefined"'
+        }
+      ],
+      validations: {
+        join_responses: true,
+        list: [
+          {
+            property: 'patient_id',
+            rule: 'len(5)',
+            message: [{
+              content: 'Invalid patient Id.',
+              locale: 'en'
+            }]
+          }
+        ]
+      }
+    }]);
+  });
+
+  it('setExpectedBirthDate sets lmp_date and expected_date correctly for 8 weeks ago', () => {
+    const doc = {
+      fields:
+      {
+        lmp_date: eightWeeksAgo.valueOf(),
+        type: 'data_record'
+      }
+    };
+
+    transition.setExpectedBirthDate(doc);
+    assert(doc.lmp_date);
+    assert.equal(doc.lmp_date, eightWeeksAgo.clone().toISOString());
+    assert.equal(doc.expected_date, eightWeeksAgo.clone().add(40, 'weeks').toISOString());
+  });
+
+  it('valid adds lmp_date and patient_id', () => {
+    sinon.stub(utils, 'getContactUuid').resolves('uuid');
+    sinon.stub(transitionUtils, 'getUniqueId').resolves(12345);
+
+    const doc = {
+      form: 'l',
+      type: 'data_record',
+      fields: {
+        lmp_date: eightWeeksAgo.valueOf(),
+        patient_name: 'abc'
+      }
+    };
+
+    return transition.onMatch({ doc: doc }).then(function (changed) {
+      assert.equal(changed, true);
+      assert.equal(doc.lmp_date, eightWeeksAgo.toISOString());
+      assert(doc.patient_id);
+      assert.equal(doc.tasks, undefined);
+    });
+  });
+
+  it('valid name invalid LMP date', () => {
+    const doc = {
+      form: 'l',
+      from: '+1234',
+      type: 'data_record',
+      fields: {
+        patient_name: 'hi',
+        lmp_date: 'x'
+      }
+    };
+
+    return transition.onMatch({ doc: doc }).then(function (changed) {
+      assert.equal(changed, true);
+      assert.equal(doc.patient_id, undefined);
+      assert.equal(getMessage(doc),
+        'Date should be later than 40 weeks ago. ' +
+        ' Date should be older than 8 weeks ago.');
+    });
+  });
+
+  it('invalid name invalid LMP Date logic', () => {
+    const doc = {
+      form: 'l',
+      from: '+123',
+      type: 'data_record',
+      fields: {
+        patient_name: '',
+        lmp_date: null
+      }
+    };
+
+    return transition.onMatch({ doc: doc }).then(function (changed) {
+      assert.equal(changed, true);
+      assert.equal(doc.patient_id, undefined);
+      assert.equal(getMessage(doc),
+        'Invalid patient name. ' +
+        ' Date should be later than 40 weeks ago. ' +
+        ' Date should be older than 8 weeks ago.');
+    });
+  });
+  
+  it('LMP date less than 8 weeks ago should fail', () => {
+    const start = moment().subtract({ weeks: 7, days: 6 }).startOf('day');
+    sinon.stub(utils, 'getContactUuid').resolves('uuid');
+    sinon.stub(transitionUtils, 'getUniqueId').resolves(12345);
+
+    const doc = {
+      form: 'l',
+      type: 'data_record',
```

Comment:
Doesn't the validation check against `reported_date`? This document should have a `reported_date` property.

Reply:
I had missed updating it after the last change. Thanks.